packages/candy_editor/src/assets/index.tsx

```diff
@@ -103,71 +119,143 @@ export const CandyDataEditor = () => {
   useEffect(() => {
     setCandyType(NOT_SELECTED);
     closeModal();
+    console.log('CANDYCLASS', candyClass);
   }, [editableCandyClass]);
 
   useEffect(() => {
     console.log('🍬 CANDYCLASS', candyClass);
   }, [candyClass]);
 
+  useEffect(() => {
+    if (candyDataEditor.existingCandyClass) {
+      setCandyClass(candyDataEditor.existingCandyClass);
+      let arrayPropertyWithId: PropertyWithId[] = [];
```

Comment:
Please refactor with map and destructuring and without hungarian notation ('array' prefix).
```
let propertiesWithId: PropertyWithId[] = candyDataEditor.existingCandyClass.Class.map((p) => {
  return { id: Math.random().toString(), ...p };
});
```

Reply:
Refactored also SAveproperty, thank you for the suggestions